src/ripple/app/tx/impl/AMMBid.cpp

```diff
@@ -0,0 +1,335 @@
+        if (ctx_.tx.isFieldPresent(sfAuthAccounts))
+            auctionSlot.setFieldArray(
+                sfAuthAccounts, ctx_.tx.getFieldArray(sfAuthAccounts));
+        else
+            auctionSlot.makeFieldAbsent(sfAuthAccounts);
+        // Burn the remaining bid amount
+        auto const saBurn = toSTAmount(lpTokens.issue(), burn);
+        if (saBurn >= lptAMMBalance)
+        {
+            JLOG(ctx_.journal.debug())
+                << "AMM Bid: invalid burn " << burn << " " << lptAMMBalance;
+            return tecAMM_FAILED_BID;
+        }
+        auto res =
+            redeemIOU(sb, account_, saBurn, lpTokens.issue(), ctx_.journal);
+        if (res != tesSUCCESS)
+        {
+            JLOG(ctx_.journal.debug()) << "AMM Bid: failed to redeem.";
+            return res;
+        }
+        ammSle->setFieldAmount(sfLPTokenBalance, lptAMMBalance - saBurn);
+        sb.update(ammSle);
+        return tesSUCCESS;
+    };
+
+    TER res = tesSUCCESS;
+
+    auto const bidMin = ctx_.tx[~sfBidMin];
+    auto const bidMax = ctx_.tx[~sfBidMax];
+
+    Number const MinSlotPrice = 0;
+
+    auto getPayPrice =
+        [&](Number const& computedPrice) -> std::optional<Number> {
+        // Both min/max bid price are defined
+        if (bidMin && bidMax)
+        {
+            if (computedPrice >= *bidMin && computedPrice <= *bidMax)
+                return computedPrice;
+            JLOG(ctx_.journal.debug())
+                << "AMM Bid: not in range " << computedPrice << *bidMin << " "
+                << *bidMax;
+            return std::nullopt;
+        }
+        // Bidder pays max(bidPrice, computedPrice)
+        if (bidMin)
+        {
+            return std::max(computedPrice, Number(*bidMin));
+        }
+        else if (bidMax)
+        {
+            if (computedPrice <= *bidMax)
+                return computedPrice;
+            JLOG(ctx_.journal.debug())
+                << "AMM Bid: not in range " << computedPrice << *bidMax;
+            return std::nullopt;
+        }
+        else
+            return computedPrice;
+    };
+
+    // No one owns the slot or expired slot.
+    if (auto const acct = auctionSlot[~sfAccount]; !acct || !validOwner(*acct))
+    {
+        if (auto const payPrice = getPayPrice(MinSlotPrice); !payPrice)
+            return {tecAMM_FAILED_BID, false};
+        else
+            res = updateSlot(0, *payPrice, *payPrice);
+    }
+    else
+    {
+        // Price the slot was purchased at.
+        STAmount const pricePurchased = auctionSlot[sfPrice];
+        assert(timeSlot);
+        auto const fractionUsed = (Number(*timeSlot) + 1) / nIntervals;
+        auto const fractionRemaining = Number(1) - fractionUsed;
+        auto const computedPrice = [&]() -> Number {
+            auto const p1_05 = Number(105, -2);
+            // First interval slot price
+            if (*timeSlot == 0)
+                return pricePurchased * p1_05 + MinSlotPrice;
```

Comment:
If `MinSlotPrice` is `0`, why do we need to add it? Can you explain the intention?

Reply:
To be consistent with how the formulas are presented in XLS30d.